app/controllers/smart_proxies_controller.rb

```diff
@@ -165,10 +165,23 @@ def resource_base
   def versions_mismatched?(proxy_versions_hash)
```

Comment:
If only the version is used, perhaps this function should only take a `proxy_version`? It could simplify testing a lot.